src/devtools/pachdev/main.go

```diff
@@ -0,0 +1,138 @@
+package main
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/spf13/cobra"
+	// logging library used by yqlib
+	logging "gopkg.in/op/go-logging.v1"
+)
+
+var verbose bool
+
+func restartClusterCmd() *cobra.Command {
+	return &cobra.Command{
+		Use:   "restart cluster",
+		Short: "Restart the entire kube cluster and redeploy Pachyderm",
+		Run: func(cmd *cobra.Command, args []string) {
```

Comment:
I end up using the Ctx variants of the Cobra command runner functions so that the context-that-has-a-logger (and signal.NotifyContext(os.Interrupt, syscall.SIGTERM) are used by all the underlying functions.  Then everything in your library that uses op.ctx can have this one passed in and the logs are clearer and cancellation from the terminal always works cleanly.  Check what src/cmd/starpach does... that is more modern code than pachctl.